stonesoup/types/association.py

```diff
@@ -66,6 +67,95 @@ def __init__(self, associations=None, *args, **kwargs):
         super().__init__(associations, *args, **kwargs)
         if self.associations is None:
             self.associations = set()
+        if not isinstance(self.associations, Set):
+            raise TypeError("Supplied parameter must be a set")
+        if not all([isinstance(member, Association) for member in self.associations]):
+            raise TypeError("Association set must contain only Association instances")
+        self._simplify()
+
+    def __eq__(self, other):
+        return True if self.associations == other.associations else False
+
+    def add(self, association):
+        if association is None:
+            return
+        elif isinstance(association, Association):
+            self.associations.add(association)
+        elif isinstance(association, AssociationSet):
+            for component in association:
+                self.add(component)
+        else:
+            raise TypeError("Supplied parameter must be an Association or AssociationSet")
+        self._simplify()
+
+    def _simplify(self):
+        """Where multiple associations describe the same pair of objects, combine them into one.
+        Note this is only implemented for pairs with a time_range attribute- others will be skipped
+        """
+        to_remove = []
```

Comment:
Not sure if this is valid use case, but should this be a set, to avoid error when trying to remove the same association twice?